h/static/scripts/directives/annotation.coffee

```diff
@@ -265,13 +265,12 @@ AnnotationController = [
           $scope.$emit('annotationUpdate')
 
       # Save highlights once logged in.
-      if highlight and this.isHighlight()
+      if this.isHighlight() and highlight
```

Comment:
There was a problem with that approach. In an earlier version of the code, I did just that and it resulted in double highlights being created. For some reason this code runs twice and you have set highlight to false so you don't create two annotations.